### ISSUE TITLE
Error when running the cmake

### ISSUE THREAD

**Issue by yonatanbitton** [snukky/news-translit-nmt#1]
Hello, this is my error:

```
`-- Could NOT find Boost (missing: timer iostreams filesystem system chrono) (found version "1.67.0")
CMake Error at CMakeLists.txt:290 (message):
  Cannot find Boost libraries.  Terminating.


-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Configuring incomplete, errors occurred!
See also "C:/Users/PC-1/Desktop/ThesisRelated/Paper/previous_works/news-translit-nmt-master/tools/marian-dev/build/CMakeFiles/CMakeOutput.log".`

```
I've tried reinstalling boost, different versions, and it doesn't seem to help. 
How can I overcome it? 

I've also opened stackoverflow post regarding this problem: (I want to run this code and cite it in my paper) 
https://stackoverflow.com/questions/57444278/cmake-doesnt-find-boost-windows

Thanks

**Comment by snukky** [comment]
Do you have installed all required Boost libraries? I don't have much experience with compiling Marian on Windows. What works fine for me is compilation within WSL (Windows Subsystem for Linux). You may also try providing a Boost path to the cmake command directly via `-DBOOST_ROOT=...`, please see https://marian-nmt.github.io/docs/#custom-boost

**Comment by yonatanbitton** [comment]
Thanks, I've just tried but received similar output:
```
`C:\Users\PC-1\Desktop\ThesisRelated\Paper\previous_works\marian\build>cmake .. -DBOOST_ROOT="C:\Program Files\boost\boost_1_67_0"
-- Project name: marian
-- Project version: v1.7.6+1d4ba732
-- Found CUDA libraries: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/lib/x64/curand.lib C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/lib/x64/cusparse.lib C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/lib/x64/cublas.lib
-- Using std::regex
-- Found OpenSSL
-- Could NOT find Boost (missing: timer iostreams filesystem system chrono) (found version "1.67.0")
CMake Error at CMakeLists.txt:315 (message):
  Cannot find Boost libraries.  Terminating.


-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Configuring incomplete, errors occurred!
See also "C:/Users/PC-1/Desktop/ThesisRelated/Paper/previous_works/marian/build/CMakeFiles/CMakeOutput.log".
`
```

I'm trying it as well in linux but receiving other problems, trying to overcome them

**Comment by snukky** [comment]
Have you seen the instructions in `marian/vs` and are they helpful? https://github.com/marian-nmt/marian/tree/master/vs

**Comment by yonatanbitton** [comment]
Yes, i've tried it now. I'm getting the following error: 
```
--Configuring incomplete, errors occurred!
CMake Error:
   Generator

      Visual Studio 15 2017 Win64

   could not find any instance of Visual Studio
```

Of course that I have Visual Studio 2019 & VS Code...

In addition, I'm trying with WSL as you suggested. 
I've installed Ubuntu 16.04 on WSL, installed gcc, g++, essentials, python, and now trying to install cuda. 

I've downloaded the relevant cuda installation file (sudo sh cuda_10.1.168_418.67_linux.run) and ran it, but getting the following error:
```
"WARNING: The file '/proc/sys/kernel/modprobe' is unavailable; the X server will use `/sbin/modprobe` as the path to the `modprobe` utility.  This path differs from the one determined by `nvidia-installer`, /bin/kmod.  Please verify that `/sbin/modprobe` works correctly or mount the /proc file system and verify that '/proc/sys/kernel/modprobe' reports the correct path.

ERROR: Unable to find the kernel source tree for the currently running kernel.  Please make sure you have installed the kernel source files for your kernel and that they are properly configured; on Red Hat Linux systems, for example, be sure you have the 'kernel-source' or 'kernel-devel' RPM installed.  If you know the correct kernel source files are installed, you may specify the kernel source path with the '--kernel-source-path' command line option."

```
To sum up what I've tried: 
1. Windows (Not finding boost, unable to supply custom boost-still not finding)
2. Linux in WSL
3. Linux in VirtualBox
Getting the same error for linux in WSL and in VirtualBox

**Comment by yonatanbitton** [comment]
This is my current status:
I built boost successfully:
```
`The Boost C++ Libraries were successfully built!

The following directory should be added to compiler include paths:

    C:\Program Files\boost\boost_1_67_0

The following directory should be added to linker library paths:

    C:\Program Files\boost\boost_1_67_0\stage\lib`

```
But when performing again `cmake .. -DBOOST_ROOT="C:\Program Files\boost\boost_1_67_0"` i'm receiving the same error again:
```
`-- Could NOT find Boost (missing: timer iostreams filesystem system chrono) (found version "1.67.0")
CMake Error at CMakeLists.txt:290 (message):
  Cannot find Boost libraries.  Terminating.


-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Configuring incomplete, errors occurred!
See also "D:/ThesisResources/previous_works/news-translit-nmt-master/tools/marian-dev/build/CMakeFiles/CMakeOutput.log".
`
```
I've tried to set variables to the compiler include paths and the linker library paths but I think I'm doing it wrong. 

I feel close to the solution, but I'm still not sure how to continue.

**Comment by yonatanbitton** [comment]
Updated status with screenshots here:
https://stackoverflow.com/questions/57444278/cmake-doesnt-find-boost-windows

**Comment by snukky** [comment]
Please keep also in mind that all scripts for reconstructing experiments have been prepared for and tested only on Unix.

**Comment by yonatanbitton** [comment]
Yes, I see. Tomorrow I'll install Ubuntu 16.04 instead of my 17.10 and try again.

**Comment by yonatanbitton** [comment]
I reinstalled Ubuntu 16.04 and the problem was solved. Still don't know how to solve it in windows, but i'll continue use Ubuntu 16.04.
